README.md

```diff
@@ -776,6 +780,169 @@ You'll see 504 gateway timeout 10 sec after you send the request
+  http:
+  - route:
+    - destination:
+        host: non-existent-srevice.testns1.svc.cluster.local 
+        port:
+          number: 80
+    retries:
+      attempts: 0
+EOF
+
+# kubectl apply -f manifests/destination-rule-connect-timeout.yaml -n testns1
+
+kubectl apply -n testns1 -f - <<EOF
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: non-existent-srevice
+spec:
+  host: non-existent-srevice.testns1.svc.cluster.local
+  trafficPolicy:
+    connectionPool:
+      tcp:
+        connectTimeout: 4s
+EOF
+```
+
+Send the HTTP request to http://non-existent-srevice.testns1.svc.cluster.local/. The endpoint is supposed to return a response after 4 sec.
+
+```bash
+export NAMESPACE=testns1
+export SLEEP_POD=$(kubectl get pod -l app=sleep -n $NAMESPACE -o jsonpath={.items..metadata.name})
+kubectl exec "${SLEEP_POD}" -c sleep -n $NAMESPACE -- curl -sv  "http://non-existent-srevice.${NAMESPACE}.svc.cluster.local/"
+```
+
+You'll see 503 timeout 4 sec after you send the request
+```txt
+* Connected to non-existent-srevice.testns1.svc.cluster.local (10.100.62.253) port 80 (#0)
+> GET / HTTP/1.1
+> Host: non-existent-srevice.testns1.svc.cluster.local
+> User-Agent: curl/7.77.0-DEV
+> Accept: */*
+>
+* Mark bundle as not supporting multiuse
+< HTTP/1.1 503 Service Unavailable
+< content-length: 84
+< content-type: text/plain
+< date: Tue, 22 Jun 2021 07:18:55 GMT
+< server: envoy
+<
+* Connection #0 to host non-existent-srevice.testns1.svc.cluster.local left intact
+```
+
+See also [ConnectionPoolSetting.TCPSettings](https://istio.io/latest/docs/reference/config/networking/destination-rule/#ConnectionPoolSettings-TCPSettings)
+
+#### Idle Timeout
+The configuration below set 5 sec http idle timeout for a request to the non-existent service.
+
+Using the service created in [Connect Timeout](#connect-timeout), delete the connectTimeout and set idleTimetou to 5 sec in DestinationRule.
+The default value of connectTimeout is 10 sec, so the request will timeout at idleTimeout 5 sec.
+
+Set the http idle timeout in spec.trafficPolicy.connectionPool.http.idleTimeout in DestinationRule.
+(The default is 1 hour.)
+```yaml
+DestinationRule
+
+# kubectl apply -f manifests/destination-rule-idle-timeout.yaml -n testns1
+
+kubectl apply -n testns1 -f - <<EOF
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: non-existent-srevice
+spec:
+  host: non-existent-srevice.testns1.svc.cluster.local
+  trafficPolicy:
+    connectionPool:
+      http:
+        idleTimeout: 5s
+EOF
+```
+
+Send the HTTP request to http://non-existent-srevice.testns1.svc.cluster.local/. The endpoint is supposed to return a response after 5 sec.
+
+```bash
+export NAMESPACE=testns1
+export SLEEP_POD=$(kubectl get pod -l app=sleep -n $NAMESPACE -o jsonpath={.items..metadata.name})
+kubectl exec "${SLEEP_POD}" -c sleep -n $NAMESPACE -- curl -sv  "http://non-existent-srevice.${NAMESPACE}.svc.cluster.local/"
```

Comment:
```suggestion
kubectl exec "${SLEEP_POD}" -c sleep -n $NAMESPACE -- curl -sv "http://non-existent-srevice.${NAMESPACE}.svc.cluster.local/"
```

Reply:
Thanks for suggestion.
Created a commit because there were several.
https://github.com/yokawasa/istio-sandbox/pull/3/commits/f524c9691c9c875edbddc408fc8d2bca98f170b1